Wikipedia/Code/WelcomeAnimationView.swift

```diff
@@ -38,4 +38,10 @@ public class WelcomeAnimationView : UIView {
     var wmf_scaleZeroAndLowerRightTransform: CATransform3D {
           return CATransform3DConcat(wmf_scaleZeroAndRightTransform, wmf_lowerTransform)
     }
+    
+    override public func didMoveToSuperview() {
+        super.didMoveToSuperview()
+        // Fix for: http://stackoverflow.com/a/39614714
+        self.layoutIfNeeded()
```

Comment:
Copy paste from the linked SO post (**emphasis mine**):

> "Sending layoutIfNeeded to a view is not expected to move the view, but in earlier releases, if the view had translatesAutoresizingMaskIntoConstraints set to NO, and if it was being positioned by constraints, layoutIfNeeded would move the view to match the layout engine **before sending layout to the subtree**. These changes correct this behavior, and the receiver’s position and usually its size won’t be affected by layoutIfNeeded.

> Some existing code may be relying on this incorrect behavior that is now corrected. There is no behavior change for binaries linked before iOS 10, but when building on iOS 10 you may need to correct some situations by sending -layoutIfNeeded to a superview of the translatesAutoresizingMaskIntoConstraints view that was the previous receiver, or else positioning and sizing it before (or after, depending on your desired behavior) layoutIfNeeded.

>  Third party apps with custom UIView subclasses using Auto Layout that override layoutSubviews and dirty layout on self before calling super are at risk of triggering a layout feedback loop when they rebuild on iOS 10. When they are correctly sent subsequent layoutSubviews calls they must be sure to stop dirtying layout on self at some point (note that this call was skipped in release prior to iOS 10)."

The constraints seem to be working fine without adding the call to `layoutSubviews` or `setFrame` - it's just the initial layout, I think, based on the way I was adding the animation views, which wasn't set to (as the comment above says) `match the layout engine` in time.